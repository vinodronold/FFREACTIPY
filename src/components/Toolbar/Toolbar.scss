@import "../../../theme_variables";
@import "@material/toolbar/mdc-toolbar";