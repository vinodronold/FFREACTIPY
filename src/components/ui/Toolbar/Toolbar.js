import React from 'react'
import './Toolbar.css'

const Toolbar = () => {
    return (
        <header className="mdc-toolbar">
            <div className="mdc-toolbar__row">
                <section className="mdc-toolbar__section mdc-toolbar__section--align-start">
                    <a className="material-icons">menu</a>
                    <span className="mdc-toolbar__title">Title</span>
                </section>
            </div>
        </header>
    )
}

export default Toolbar