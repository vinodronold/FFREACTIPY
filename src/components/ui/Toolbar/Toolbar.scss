@import "../../../vars.scss";
@import "@material/toolbar/mdc-toolbar.scss";