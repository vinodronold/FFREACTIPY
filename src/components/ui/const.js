export const materialIcon = "material-icons"