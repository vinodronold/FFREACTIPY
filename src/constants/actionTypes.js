export const TOGGLE_DRAWER = 'TOGGLE_DRAWER'
export const TOGGLE_MORE_OPTIONS = 'TOGGLE_MORE_OPTIONS'
export const UPDATE_HEADER = 'UPDATE_HEADER'

export const YT_PLAYER_STATUS_CHANGED = 'YT_PLAYER_STATUS_CHANGED'

export const YT_PLAYER_PLAYING = 'YT_PLAYER_PLAYING'
export const YT_PLAYER_BUFFERING = 'YT_PLAYER_BUFFERING'
export const YT_PLAYER_PAUSED = 'YT_PLAYER_PAUSED'
export const YT_PLAYER_ENDED = 'YT_PLAYER_ENDED'
export const YT_PLAYER_CUED = 'YT_PLAYER_CUED'