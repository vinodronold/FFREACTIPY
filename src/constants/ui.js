export const materialIcon = 'material-icons'
export const clsButton = 'mdc-button'
export const clsElevation = 'mdc-elevation'
export const clsToolbar = 'mdc-toolbar'
export const clsTempDrawer = 'mdc-temporary-drawer'
export const clsList = 'mdc-list'
export const clsTypography = 'mdc-typography'
export const clsTextField = 'mdc-textfield'
export const clsTheme = 'mdc-theme'