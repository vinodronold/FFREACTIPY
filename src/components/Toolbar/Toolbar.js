import React from 'react'
import classNames from 'classnames'
import { materialIcon, clsToolbar } from '../../constants/ui'
import './Toolbar.css'

const Toolbar = ({ onMenuClick }) => {
    return (
        <header className={classNames([`${clsToolbar}`, `${clsToolbar}--fixed`])}>
            <div className={`${clsToolbar}__row`}>
                <section className={classNames([`${clsToolbar}__section`, `${clsToolbar}__section--align-start`])}>
                    <a className={materialIcon} onClick={onMenuClick}>menu</a>
                </section>
                <section
                    className={classNames([
                        `${clsToolbar}__section`,
                        `${clsToolbar}__section--align-end`,
                        `${clsToolbar}__section--shrink-to-fit`
                    ])}>
                    <a className={materialIcon} aria-label="Search" alt="Search">search</a>
                </section>
            </div>
        </header>
    )
}

export default Toolbar
