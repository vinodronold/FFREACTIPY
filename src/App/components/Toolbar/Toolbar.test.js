import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Toolbar from './Toolbar'
import { clsToolbar } from '../../constants/ui'

describe('<Toolbar />', () => {
    const testToolbar = shallow(<Toolbar />)
    it('should have one header', () => {
        expect(testToolbar.find('header')).toHaveLength(1)
    })    
    it('should have header with fixed class - ' + `header.${clsToolbar}--fixed`, () => {
        expect(testToolbar.find(`header.${clsToolbar}--fixed`)).toHaveLength(1)
    })    
    it('should have one row class - ' + `div.${clsToolbar}__row`, () => {
        expect(testToolbar.find(`div.${clsToolbar}__row`)).toHaveLength(1)
    })    
    it('should have two sections', () => {
        expect(testToolbar.find('section')).toHaveLength(2)
    })    
    it('should match the Last Snapshot', () => {
        expect(toJson(testToolbar)).toMatchSnapshot()
    })
})