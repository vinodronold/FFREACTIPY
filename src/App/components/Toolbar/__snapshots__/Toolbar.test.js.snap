exports[`<Toolbar /> should match the Last Snapshot 1`] = `
<header
  className="mdc-toolbar mdc-toolbar--fixed">
  <div
    className="mdc-toolbar__row">
    <section
      className="mdc-toolbar__section mdc-toolbar__section--align-start">
      <a
        className="material-icons">
        menu
      </a>
    </section>
    <section
      className="mdc-toolbar__section mdc-toolbar__section--align-end mdc-toolbar__section--shrink-to-fit">
      <a
        alt="Search"
        aria-label="Search"
        className="material-icons">
        search
      </a>
    </section>
  </div>
</header>
`;
